### PR TITLE
add flag for reading as 32-bit

### LINKED ISSUE #86: Cannot access blob data inserted with older version of `mym`
### Description
Cannot open blob fields that are inserted with an old version of `mym` with either a newer version of `mym` or `datajoint-python`>=0.13.

### Reproducibility

Access with `mym` version 1.36 works well.
- OS  Windows
- MATLAB 2019b
- MySQL 5.1.73-log
- mym 1.36 (hosted [here](https://github.com/Brody-Lab/ExperPort/tree/master/MySQLUtility))

```
mym('version')
Matlab<->MySQL connector v1.36
data = mym('select protocol_data from bdata.sessions where sessid=816988')

data =

  struct with fields:

    protocol_data: {[1×1 struct]}
```

Access with `mym` version 2.8.2 does not work.
- OS  MAC
- MATLAB 2020a
- MariaDB 10.2.33 (This is a copy of the tables used on the original database.)
- mYm 2.8.2

```
mym('version')
mYm version 2.8.2
data = mym('select protocol_data from bdata.sessions where sessid=816988')
Out of memory.
```

Access with `datajoint-python` version 0.13.2 does not work.
- OS  MAC
- MariaDB 10.2.33 (This is a copy of the tables used on the original database.)
- DataJoint Python 0.13.2

Retrieves a byte string that am not sure how to decode.
```
subject_session_key = {'sessid': 816986}
session_data = (bdata.Sessions & subject_session_key).fetch('protocol_data', as_dict=True)

[{'protocol_data': b'ZL123\x007\x04\x00\x00x\x9c\xcb\x8d\xcce\x08fb```\x84b> .\xceLI-f\xc8\xc8,a\xc8\xc9OfH.(a\x08(\xca\xf7\x89O\xce\x033\x82\xc0\x0c\xc7\xbc\x12\x1f\x18#\x08&\xe5\x13_\x90Z\x94\x06Q\x04f\x81U\xc1XP1\xdf\xcc<?Og\x06\xdf\xc4\n\x10\xe5\t\xb4\xcf\x11\xe4\x00)\xa8\x03X\x18  \x87\x0c\xf8\x14\xdd06\x06R\xc0\x07{T\x1a]\x1c\x97zB|\\4\xb1\xea0\xf5\xe1\x0c\xb5\x022 \x81P\xfb\xf1\x7f\xb8\xd0\xb20\x8f\xc2\x12;\xaaG\xad\x1c\x08(``\xa0\xa1\x82$\xf52 zdO\xc0\x04\x82\xbe\xa0D\xc1\x9f\x95\x1f/\xf9&\x05\xe0v\x03L\x01\x00w"\x9b\xf2'}]
```

Thanks!

### PR DESCRIPTION
**Pull request by A-Baji**
fix #86

### CODE REVIEW

**Comment by A-Baji** [comment]
The value for the repository secret `MATLAB_LICENSE_R2018B` needs to be updated to the same as `MATLAB_LICENSE_R2019A`.

I think alternatively, dynamically setting the license is unnecessary since the license for R2019A works for both R2018B and R2016B versions, so a single secret `MATLAB_LICENSE` value would work.

File: tests/TestExternal.m [hunk 1]
```diff
@@ -90,6 +90,24 @@ function TestExternal_testdj0Error(testCase)
...
+            % https://github.com/datajoint/mym/issues/86
+            value = ['6D596D00530200000001000000010000000400000068697473007369646573007461736B73007374616765004D000000410200000001000000070000000600000000000000000000000000F8FF000000000000F03F000000000000F03F0000000000000000000000000000F03F0000000000000000000000000000F8FF230000004102000000010000000700000004000000000000006C006C006C006C00720072006C002300000041020000000100000007000000040000000000000064006400640064006400640064002500000041020000000100000008000000040000000000000053007400610067006500200031003000'];
+            hexstring = value';
+            reshapedString = reshape(hexstring,2,length(value)/2);
+            hexMtx = reshapedString.';
+            decMtx = hex2dec(hexMtx);
+            packed = uint8(decMtx);
+            try
+                setenv('USE_32BIT_DIMS', 'true');
+                unpacked = mym('deserialize', packed);
```
Comment (guzman-raphael) [inline tests/TestExternal.m]: Could you update your test to verify the contents of the unpacked data?

File: tests/TestExternal.m [hunk 2]
```diff
@@ -90,6 +90,24 @@ function TestExternal_testdj0Error(testCase)
...
+                setenv('USE_32BIT_DIMS', 'false');
+            catch ME
+                testCase.verifyEqual(ME.identifier, 'mYm:CrossPlatform:Compatibility');
+            end
```
Comment (guzman-raphael) [inline tests/TestExternal.m]: We can remove the `try/catch` here since we are not checking for an explicit error message being thrown.

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
@A-Baji Glad to see you helping out with `mYm`! :handshake: Great work!

Could you also update a few more items?
- bump the version in `mym.h`
- update the binaries to reflect these changes for Linux, Windows, and MacOS (see `./mex_compilation`)
- Dimitri had a good [comment](https://github.com/datajoint/mym/pull/89#discussion_r814960636) that I think was missed

File: src/mym.cpp [hunk 3]
```diff
@@ -353,6 +358,17 @@ static void updateplugindir() {
...
+/**********************************************************************
+ *use32bitdims():   Check if dimensions should be read as 32-bit
+ *  Get USE_32_BIT_DIMS environment variable
+ *  Return true/false based on USE_32_BIT_DIMS flag
+ **********************************************************************/
+static bool use32bitdims() {
+    if (getenv("USE_32BIT_DIMS"))
+        return strcasecmp(getenv("USE_32BIT_DIMS"), "true") == 0;
+    else
+        return false;
```
Comment (guzman-raphael) [inline src/mym.cpp]:
```suggestion
    return getenv("USE_32BIT_DIMS") && strcasecmp(getenv("USE_32BIT_DIMS"), "true") == 0;
```

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
@A-Baji Think this should satisfy @dimitri-yatsenko's earlier point about simplification.

Also, noticed that we don't have a test to verify the version of the binary. Could you add one in as well? Let's bump it to `2.8.4` to help us easily verify the new binaries.

We should be good-to-go after that and I can include the binaries for other OS'.

**Comment by A-Baji** [comment]
> Also, noticed that we don't have a test to verify the version of the binary. Could you add one in as well?

Could you elaborate on this? Does it have anything to do with the binary version check in the linux docker compose?

**Comment by guzman-raphael** [comment]
@A-Baji Ah great point! Forgot that we do display the binary version already. That is enough. Think we are all good now and thanks for the updates. I'll try to add binaries for the other OS' around 8AM CST tomorrow. We can merge after that.

**Review by guzman-raphael — APPROVED** [review]

**Comment by guzman-raphael** [comment]
@A-Baji Excellent work and great to collab with you on this one! :handshake:

**Pull request merged**